tests/util.py

```diff
@@ -0,0 +1,171 @@
+from dataclasses import dataclass
+from datetime import date
+from google.cloud import bigquery
+from typing import Any, Callable, Dict, List, Optional, Set
+
+import json
+import os.path
+import yaml
+
+table_extensions = {
+    "ndjson": bigquery.SourceFormat.NEWLINE_DELIMITED_JSON,
+    "csv": bigquery.SourceFormat.CSV,
+    "backup_info": bigquery.SourceFormat.DATASTORE_BACKUP,
+    "export_metadata": bigquery.SourceFormat.DATASTORE_BACKUP,
+    "avro": bigquery.SourceFormat.AVRO,
+    "parquet": bigquery.SourceFormat.PARQUET,
+    "orc": bigquery.SourceFormat.ORC,
+}
+
+
+@dataclass(frozen=True)
+class Table:
+    name: str
+    source_format: str
+    source_path: str
+
+    def get_schema(self):
+        resource_dir, resource = os.path.split(self.source_path)
+        full_name, _ = resource.rsplit(".", 1)
+        try:
+            return [
+                bigquery.SchemaField.from_api_repr(field)
+                for field in load(resource_dir, f"{full_name}.schema")
+            ]
+        except FileNotFoundError:
+            return None
+
+
+@dataclass
+class GeneratedTest:
+    expect: List[Dict[str, Any]]
+    name: str
+    query: str
+    query_name: str
+    query_params: List[Any]
+    replace: Dict[str, str]
+    tables: Set[Table]
```

Comment:
using `Set[Table]` isn't really better than `List[Table]` or perhaps `Dict[str, Table]`, and requires table to be frozen, which makes Table.get_schema more complicated than necessary. Should probably switch to `list` or `dict`.